ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/anomalylocalization/AnomalyLocalizerImplTests.java

```diff
@@ -200,8 +233,14 @@ public void setup() {
         expectedOutput.getResults().put(agg.getName(), result);
     }
 
+    @AfterClass
+    public static void resetPortCounter() {
+        portGenerator.set(0);
+    }
+
     @Test
     public void testGetLocalizedResultsGivenNoAnomaly() {
+        when(clusterService.state()).thenReturn(testState);
```

Comment:
Did you just add this line to resolve the "null" problem?

Reply:
yes, I was confused before about what should the mocked ClusterService return. I thought it should return a mocked local node, but turns out it's true